os/board/common/crashdump.c

```diff
@@ -38,13 +41,26 @@
  * Definitions
  ****************************************************************************/
 #if defined(CONFIG_BOARD_RAMDUMP_UART)
-#define HANDSHAKE_STRING                "RAMDUMP"
-#define HANDSHAKE_STR_LEN_MAX           (7)
+#define HANDSHAKE_STRING		"RAMDUMP"
+#define HANDSHAKE_STR_LEN_MAX		(7)
 #endif
 
+#if defined(CONFIG_BOARD_COREDUMP_FLASH)
+#define MTD_PARTNAME_LEN		15
+#define COREDUMP_BUF_SIZE		(4096)
+#endif
+
+#if defined(CONFIG_BOARD_RAMDUMP_FLASH)
```

Comment:
Is it possible to enable all, CONFIG_BOARD_RAMDUMP_UART, CONFIG_BOARD_COREDUMP_FLASH and CONFIG_BOARD_RAMDUMP_FLASH? I think at least UART and FLASH could not enable at the same time.
If it is, you need to use `else if`

Reply:
we can enable flash write and uart read same time, so we will maintain current change now